File: apps/engine/tests/units/multiple_groups/sugarscape/dependencies/@hash/age/src/behaviors/age.js
/**
 * @param {AgentState} state of the agent
 * @param {AgentContext} context of the agent
 */
const behavior = (state, context) => {
  if (state.age) {
    state.age += 1;
  } else {
    state.age = 1;
  }
};

File: apps/engine/tests/examples/ant_foraging/dependencies/@hash/age/src/behaviors/age.rs
use crate::prelude::{AgentState, Context, SimulationResult};

pub fn age(mut state: AgentState, _context: &Context) -> SimulationResult<AgentState> {
    let age = match state["age"].as_i64() {
        Some(age) => age + 1,
        None => 1,
    };

    state["age"] = json!(age);

    Ok(state)
}
